src/Sylius/Bundle/AdminBundle/Twig/OrderSummaryExtension.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Twig;
+
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class OrderSummaryExtension extends AbstractExtension
+{
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction(
+                'sylius_order_summary_unit_order_discount', [$this, 'getUnitOrderDiscount']
+            ),
+            new TwigFunction(
+                'sylius_order_summary_discounted_unit_price_and_quantity', [$this, 'getDiscountedUnitPriceAndQuantity']
+            ),
+            new TwigFunction(
+                'sylius_order_summary_subtotal', [$this, 'getSubtotal']
+            ),
+        ];
+    }
+
+    public function getUnitOrderDiscount(OrderItemInterface $orderItem): array
+    {
+        $units = $orderItem->getUnits();
+
+        $unitsTypes = [];
```

Comment:
if I good understand these functions  `$unitTypes` contain associative array price => quantity so `$unitsTypes` should by named better, because the current name, don't describe properly the content of the array.